airflow/api_connexion/endpoints/dataset_endpoint.py

```diff
@@ -124,3 +130,153 @@ def get_dataset_events(
     return dataset_event_collection_schema.dump(
         DatasetEventCollection(dataset_events=events, total_entries=total_entries)
     )
+
+
+def _generate_queue_event_where_clause(
+    *,
+    dag_id: str | None = None,
+    dataset_id: int | None = None,
+    uri: str | None = None,
+    before: str | None = None,
+) -> list:
+    """Get DatasetDagRunQueue where clause."""
+    where_clause = []
+    if dag_id is not None:
+        where_clause.append(DatasetDagRunQueue.target_dag_id == dag_id)
+    if dataset_id is not None:
+        where_clause.append(DatasetDagRunQueue.dataset_id == dataset_id)
+    if uri is not None:
+        where_clause.append(DatasetModel.uri == uri)
+    if before is not None:
+        where_clause.append(DatasetDagRunQueue.created_at < format_datetime(before))
+    return where_clause
+
+
+@security.requires_access_dataset("GET")
```

Comment:
```suggestion
@security.requires_access_dataset("GET")
@security.requires_access_dag("GET")
```

We should probably also check the user has read on the DAG.

Reply:
Sure! Just added it. Thanks!